src/main/java/com/solvd/automation/lab/carina/demo/gui/pages/GSMArenaHomePage.java

```diff
@@ -0,0 +1,170 @@
+    private ExtendedWebElement searchBar;
+    @FindBy(className = "tip-icon")
+    private ExtendedWebElement tipIcon;
+    @FindBy(className = "fb-icon")
+    private ExtendedWebElement fbIcon;
+    @FindBy(className = "tw-icon")
+    private ExtendedWebElement twIcon;
+    @FindBy(className = "ig-icon")
+    private ExtendedWebElement igIcon;
+    @FindBy(className = "yt-icon")
+    private ExtendedWebElement ytIcon;
+    @FindBy(className = "rss-icon")
+    private ExtendedWebElement rssIcon;
+    @FindBy(id = "login-active")
+    private ExtendedWebElement loginIcon;
+    @FindBy(xpath = "//a[@href='register.php3']")
+    private ExtendedWebElement signUp;
+    @FindBy(className = "icon-count")
+    private ExtendedWebElement loginIconOfRegisteredUser;
+
+    @FindBy(id = "login-popup2")
+    private ExtendedWebElement loginFormParent;
+    @FindBy(xpath = "//p[text()='Login']")
+    private ExtendedWebElement loginTitle;
+    @FindBy(id = "email")
+    private ExtendedWebElement emailInput;
+    @FindBy(id = "upass")
+    private ExtendedWebElement passwordInput;
+    @FindBy(id = "nick-submit")
+    private ExtendedWebElement submitButton;
+    @FindBy(partialLinkText = "I forgot my password")
+    private ExtendedWebElement forgotPasswordLink;
+    @FindBy(xpath = "//h3[text()='Login failed.']")
+    private ExtendedWebElement wrongPasswordLink;
+    @FindBy(xpath = "//p[text()='Reason: Wrong password.']")
+    private ExtendedWebElement wrongPasswordReasonLink;
+    @FindBy
+    List<ExtendedWebElement> phoneLink;
+
+    @FindBy(xpath = "//h3[text()='Login successful.']")
+    private ExtendedWebElement loginSuccessMessage;
+
+
+    public List<ExtendedWebElement> heder =
+            Arrays.asList(searchBar, tipIcon,
+                    twIcon, fbIcon, twIcon, igIcon,
+                    ytIcon, rssIcon, loginIcon, signUp);
+
+    public GSMArenaHomePage(WebDriver driver) {
+        super(driver);
+    }
+
+    @Override
+    public boolean isPageOpened() {
+
+        return logo.isElementPresent();
+    }
+
+    public boolean validateBaseElementsOnPageHeader() {
+        SoftAssert softAssert = new SoftAssert();
+        heder.forEach(locator ->
+                softAssert.assertNotNull(
+                        locator.isPresent(),
+                        String.format("%s is not found on the page.", locator.toString()
+                        )
+                )
+        );
+        softAssert.assertAll();
+        return true;
+    }
+
+    public GSMPhonePage openPhonePageFromGrid(int i) {
+        phoneLink.get(i).click();
+        return new GSMPhonePage(driver);
+    }
+
+    public GSMArenaSignUpPage openGSMArenaSignUpPage() {
+        signUp.click();
+        return new GSMArenaSignUpPage(driver);
+    }
+
+    public GSMArenaHomePage checkLoginForm() {
+        loginIcon.click();
+        return this;
+    }
+
+    public GSMArenaHomePage userLoginFault() {
+        emailInput.type(TestData.INCORRECT_LOGIN);
```

Comment:
do not define test logic in page class. define the common method like `login(String userName, String password)` and pass the corresponding arguments int the test